tests/integration/e2e/helpers.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from typing import Dict
+
+from juju.unit import Unit
+from pymongo import MongoClient
+
+logger = logging.getLogger()
+
+
+def check_produced_and_consumed_messages(uris: str, collection_name: str):
+    """Check that messages produced and consumed are consistent."""
+    logger.debug(f"MongoDB uris: {uris}")
+    logger.debug(f"Topic: {collection_name}")
+    produced_messages = []
+    consumed_messages = []
+    try:
+        client = MongoClient(
+            uris,
+            directConnection=False,
+            connect=False,
+            serverSelectionTimeoutMS=1000,
+            connectTimeoutMS=2000,
+        )
+        db = client[collection_name]
+        consumer_collection = db["consumer"]
+        producer_collection = db["producer"]
+
+        logger.info(f"Number of messages from consumer: {consumer_collection.count_documents({})}")
+        logger.info(f"Number of messages from producer: {producer_collection.count_documents({})}")
+        assert consumer_collection.count_documents({}) > 0
+        assert producer_collection.count_documents({}) > 0
+
+        cursor = consumer_collection.find({})
+        for document in cursor:
+            consumed_messages.append((document["origin"], document["content"]))
+
+        cursor = producer_collection.find({})
+        for document in cursor:
+            produced_messages.append((document["origin"], document["content"]))
```

Comment:
**suggestion** I believe one could have used the "_id" (see [here](https://github.com/canonical/kafka-test-app/blob/main/lib/charms/kafka/v0/client.py#L390)) 

```
len(producer_collection.distinct("_id"))
```